exporters/memory/include/opentelemetry/exporters/memory/in_memory_metric_exporter.h

```diff
@@ -0,0 +1,79 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <atomic>
+#include <vector>
+
+#include "opentelemetry/exporters/memory/in_memory_data.h"
+#include "opentelemetry/sdk/metrics/export/metric_producer.h"
+#include "opentelemetry/sdk/metrics/instruments.h"
+#include "opentelemetry/sdk/metrics/push_metric_exporter.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace memory
+{
+
+using InMemoryMetricData = InMemoryData<sdk::metrics::ResourceMetrics>;
```

Comment:
Define InMemoryMetricData with a dedicated class in a dedicated header file.

This is about the SDK ABI, an application may depends on InMemoryMetricData, but _not_ depend on the exporter actual implementation.

Reply:
Done.